Mesh_3/include/CGAL/exude_mesh_3.h

```diff
@@ -25,68 +25,132 @@
+    return exude_mesh_3_impl(c3t3,time_limit,sliver_bound);
+}
 
-#if defined(BOOST_MSVC)
-#  pragma warning(push)
-#  pragma warning(disable:4003) // not enough actual parameters for macro
+#ifndef CGAL_NO_DEPRECATED_CODE
+template<typename C3T3, typename CGAL_NP_TEMPLATE_PARAMETERS>
+Mesh_optimization_return_code exude_mesh_3(C3T3& c3t3, double time_limit = 0, double sliver_bound = 0)
+{
+  return exude_mesh_3(c3t3, CGAL::parameters::time_limit(time_limit).sliver_bound(sliver_bound));
+}
 #endif
-
-// see <CGAL/config.h>
-CGAL_PRAGMA_DIAG_PUSH
-// see <CGAL/boost/parameter.h>
-CGAL_IGNORE_BOOST_PARAMETER_NAME_WARNINGS
-
-BOOST_PARAMETER_FUNCTION(
-  (Mesh_optimization_return_code),
-  exude_mesh_3,
-  parameters::tag,
-  (required (in_out(c3t3),*) )
-  (optional
-    (time_limit_, *, 0 )
-    (sliver_bound_, *, parameters::default_values_for_mesh_3::exude_sliver_bound )
-  )
-)
+#ifndef DOXYGEN_RUNNING
+#if !defined(BOOST_MSVC) || (BOOST_MSVC >= 1910)
+template<typename C3T3, typename ... CGAL_NP_TEMPLATE_PARAMETERS_VARIADIC>
+Mesh_optimization_return_code exude_mesh_3(C3T3& c3t3, const CGAL_NP_CLASS& ... nps)
 {
-  return exude_mesh_3_impl(c3t3, time_limit_, sliver_bound_);
+  return exude_mesh_3(c3t3,internal_np::combine_named_parameters(nps...));
+}
+#else
+template<typename C3T3, typename CGAL_NP_TEMPLATE_PARAMETERS_NO_DEFAULT_1, typename CGAL_NP_TEMPLATE_PARAMETERS_NO_DEFAULT_2, typename ... NP>
+Mesh_optimization_return_code exude_mesh_3(C3T3& c3t3, const CGAL_NP_CLASS_1&  np1, const CGAL_NP_CLASS_2&  np2, const NP& ... nps)
+{
+  return exude_mesh_3(c3t3,internal_np::combine_named_parameters(np1, np2, nps...));
 }
-CGAL_PRAGMA_DIAG_POP
-
-#if defined(BOOST_MSVC)
-#  pragma warning(pop)
 #endif
 
 
 
+
 template <typename C3T3>
 Mesh_optimization_return_code
 exude_mesh_3_impl(C3T3& c3t3,
-                  const double time_limit,
-                  const double sliver_bound)
+                const double time_limit,
+                const double sliver_bound)
 {
-  typedef typename C3T3::Triangulation Tr;
-  typedef Mesh_3::Min_dihedral_angle_criterion<Tr> Sc;
-  //typedef Mesh_3::Radius_radio_criterion<Tr> Sc;
-  typedef typename Mesh_3::Slivers_exuder<C3T3, Sc> Exuder;
+   typedef typename C3T3::Triangulation Tr;
+   typedef Mesh_3::Min_dihedral_angle_criterion<Tr> Sc;
+   //typedef Mesh_3::Radius_radio_criterion<Tr> Sc;
+   typedef typename Mesh_3::Slivers_exuder<C3T3, Sc> Exuder;
 
-  // Create exuder
-  Sc criterion(sliver_bound, c3t3.triangulation());
-  Exuder exuder(c3t3, criterion);
+   // Create exuder
+   Sc criterion(sliver_bound, c3t3.triangulation());
+   Exuder exuder(c3t3, criterion);
 
-  // Set time_limit
-  exuder.set_time_limit(time_limit);
+   // Set time_limit
+   exuder.set_time_limit(time_limit);
 
-  // Launch exudation
-  return exuder();
+   // Launch exudation
+   return exuder();
 }
-
+#endif //DOXYGEN_RUNNING
 
 } //namespace CGAL
 
-#include <CGAL/enable_warnings.h>
```

Comment:
Disabled warnings, but not re-enabling them (other places too, just check - remove? - `disable_warnings.h` inclusions)

Reply:
I checked that they are all consistent.